### ISSUE TITLE
Investigation of eth stark

### ISSUE THREAD

**Issue by ucwong** [CortexFoundation/CortexTheseus#628]
A new ZKP library for ethereum
```
1. Code review
2. How to use
3. Run some examples
4. Advantages with others
```

https://github.com/starkware-libs/ethSTARK

**Comment by ucwong** [comment]
https://github.com/CortexFoundation/CortexTheseus/issues/630

**Comment by DhunterAO** [comment]
They claims that ethSTARK code will prove statements 20X faster than any other zero-knowledge proof code.
![image](https://user-images.githubusercontent.com/13730168/86757825-ff96c900-c075-11ea-8145-06d5b0936782.png)

**Comment by ucwong** [comment]
StarkWare (@StarkWareLtd) Tweeted:
In 2018 @ethereum gave us a grant to select a STARK-friendly hash function & develop open-source ZKP code. https://t.co/mmlLBe7wMK 
We will deliver our work (pre-audit) in a few weeks, ahead of schedule. 
ethSTARK code will prove statements 20X faster than any ZKP code out there https://t.co/8mZUKI5wGI https://twitter.com/StarkWareLtd/status/1264911004099543040?s=20 @DhunterAO

**Comment by ucwong** [comment]
It will be good if they can hold both speed and security.

**Comment by DhunterAO** [comment]
I havn't find the description about the proof size and verification time cost. It is not as good as zkSnark in my memory.

**Comment by ucwong** [comment]
> I havn't find the description about the proof size and verification time cost. It is not as good as zkSnark in my memory. They only claims about the proving speed.

I think they will be less in security.

**Comment by ucwong** [comment]
In my opinion, zokrates is a good toolbox for zkp and ethereum smart contract

**Comment by DhunterAO** [comment]
> In my opinion, zokrates is a good toolbox for zkp and ethereum smart contract, it will be easy for you when you want to use zkp to do something on blockchain

Yeah, I agree with that. I also think it does matter about the size of proof, because the space is limited on blockchain.
